### PR TITLE
zypper_lifecycle: Use select_(user_)?serial_terminal

### PR DESCRIPTION
**Pull request by dzedro**
- Related ticket: https://progress.opensuse.org/issues/126794 https://github.com/os-autoinst/os-autoinst/pull/2295
- Verification run: https://openqa.suse.de/tests/overview?build=lifecycle&version=15-SP5&distri=sle

### CODE REVIEW

**Review by b10n1k — APPROVED** [review]

**Review by grisu48 — APPROVED** [review]

**Comment by rfan1** [comment]
One minor comment, the test case is still run on xen and hyper-v
https://openqa.suse.de/tests/overview?arch=&flavor=&machine=&test=&modules=zypper_lifecycle&module_re=&distri=sle&version=15-SP5&build=88.1&groupid=110#
Can you please do VRs as well?

**Comment by dzedro** [comment]
> One minor comment, the test case is still run on xen and hyper-v https://openqa.suse.de/tests/overview?arch=&flavor=&machine=&test=&modules=zypper_lifecycle&module_re=&distri=sle&version=15-SP5&build=88.1&groupid=110# Can you please do VRs as well?

Will do, have to fix some issue happening on s390x too.

**Comment by pevik** [comment]
@b10n1k @grisu48 @rfan1 FYI this will probably be solved by https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17306, but to be merged we need first https://github.com/os-autoinst/os-autoinst/pull/2330 be deployed to the all workers.

File: lib/susedistribution.pm [hunk 1]
```diff
@@ -901,7 +901,7 @@ sub activate_console {
         assert_screen "text-logged-in-$user", 60;
     }
     elsif ($type eq 'serial-ssh') {
-        serial_terminal::set_serial_prompt($user eq 'root' ? '# ' : '$ ');
+        serial_terminal::set_serial_prompt($user eq 'root' ? '# ' : '> ');
```
Comment (mdoucha) [inline lib/susedistribution.pm:904]: What is the non-root prompt change for, exactly?

**Comment by jlausuch** [comment]
@dzedro 
This seems to break some tests:
- https://openqa.suse.de/tests/11554027#step/zypper_lifecycle/2
- https://openqa.suse.de/tests/11553977#step/zypper_lifecycle/2
- https://openqa.suse.de/tests/11553973#step/zypper_lifecycle/11
- https://openqa.suse.de/tests/11556887#step/zypper_lifecycle/11

**Comment by dzedro** [comment]
> @dzedro This seems to break some tests:
> 
>     * https://openqa.suse.de/tests/11554027#step/zypper_lifecycle/2
> 
>     * https://openqa.suse.de/tests/11553977#step/zypper_lifecycle/2
> 
>     * https://openqa.suse.de/tests/11553973#step/zypper_lifecycle/11
> 
>     * https://openqa.suse.de/tests/11556887#step/zypper_lifecycle/11

https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17405

File: lib/susedistribution.pm (hunk 1 above)
Comment (dzedro) [reply lib/susedistribution.pm:904]: Generally user console has > not $
https://openqa.suse.de/tests/11555360/logfile?filename=serial_terminal_user.txt
I had some failures related to that, but can't find them anymore.

Comment (mdoucha) [reply lib/susedistribution.pm:904]: User console will have whatever prompt you pass to `set_serial_prompt()`. If you pass `$ `, it'll have `$ `. If you pass `> `, it'll have `> `. `set_serial_prompt()` changes the shell prompt by setting `$PS1`.

**Comment by dzedro** [comment]
@jlausuch I revert also the serial console, there are some issues on JeOS and s390x, also will remove the multiline.
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17412

File: lib/susedistribution.pm (hunk 1 above)
Comment (dzedro) [reply lib/susedistribution.pm:904]: But whatever will not work so well on s390x https://openqa.suse.de/tests/11567269#step/zypper_lifecycle/127
I don't see reason to change default `>` to `$`

Comment (mdoucha) [reply lib/susedistribution.pm:904]: That failure is a mismatch between
https://github.com/dzedro/os-autoinst-distri-opensuse/blob/bc5353283bcd1647d7fc799e0316eee1ecb2d730/lib/serial_terminal.pm#L365
and
https://github.com/dzedro/os-autoinst-distri-opensuse/blob/bc5353283bcd1647d7fc799e0316eee1ecb2d730/lib/susedistribution.pm#L860

**Pull request merged**